ambari-server/src/main/java/org/apache/ambari/server/controller/internal/StackDependencyResourceProvider.java

```diff
@@ -81,6 +83,7 @@
       .put(Resource.Type.StackService, DEPENDENT_SERVICE_NAME_ID)
       .put(Resource.Type.StackServiceComponent, DEPENDENT_COMPONENT_NAME_ID)
       .put(Resource.Type.StackServiceComponentDependency, COMPONENT_NAME_ID)
+      .put(Resource.Type.StackServiceComponentDependencyType, TYPE_ID)
```

Comment:
I'm not sure what these key properties are used for, but: should type really be a part of the key?  I would think each (service+component, dependent service+component) pair to be unique, regardless of type.  No pair should have both inclusive and exclusive dependency at the same time.